### PR TITLE
Added Event Types

### PR DESCRIPTION
**Pull request by zsweigart**
Adds types for onEvent callbacks

### CODE REVIEW

File: types/LinkEventTypes.d.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+export type LinkEventListener = (linkEvent: LinkEvent) => void
+
+ 
+export interface LinkEvent {
+  eventName: LinkEventName;
+  metadata: LinkEventMetadata;
+}
+ 
+export interface LinkEventMetadata {
```
Comment (JeroenMols) [inline types/LinkEventTypes.d.ts:9]: This has `snake_case` vs `LinkEvent` that has `camelCase`

File: types/LinkEventTypes.d.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  error_type: string;
+  exit_status: string;
+  institution_id: string;
+  institution_name: string;
+  institution_search_query: string;
+  request_id: string;
+  link_session_id: string;
+  mfa_type: string;
+  view_name: LinkEventViewName;
+  timestamp: string;
```
Comment (JeroenMols) [inline types/LinkEventTypes.d.ts:21]: Order should be:

1. sessionID
2. mfatype
3. request id
4. timestamp
5. view name

File: types/LinkEventTypes.d.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  MATCHED_CONSENT = 'matched_consent',
+  MATCHED_SELECT_INSTITUTION = 'matched_select_institution',
+  OPEN = 'open',
+  OPEN_MY_PLAID = 'open_my_plaid',
+  OPEN_OAUTH = 'open_oauth',
+  SEARCH_INSTITUTION = 'search_institution',
+  SELECT_INSTITUTION = 'select_institution',
+  SUBMIT_CREDENTIALS = 'submit_credentials',
+  SUBMIT_MFA = 'submit_mfa',
+  TRANSITION_VIEW = 'transition_view',
```
Comment (JeroenMols) [inline types/LinkEventTypes.d.ts:39]: Missing unknown type

File: types/LinkEventTypes.d.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  CONSENT = 'CONSENT',
+  CREDENTIAL = 'CREDENTIAL',
+  ERROR = 'ERROR',
+  EXIT = 'EXIT',
+  LOADING = 'LOADING',
+  MFA = 'MFA',
+  NUMBERS = 'NUMBERS',
+  RECAPTCHA = 'RECAPTCHA',
+  SELECT_ACCOUNT =  'SELECT_ACCOUNT',
+  SELECT_INSTITUTION = 'SELECT_INSTITUTION',
```
Comment (JeroenMols) [inline types/LinkEventTypes.d.ts:53]: missing unknown type

**Review by JeroenMols — COMMENTED** [review]
Please look into casing and policy on #types per file

File: types/Types.d.ts [hunk 5]
```diff
@@ -315,3 +315,58 @@ export enum LinkErrorType {
     RECAPTCHA_ERROR = 'RECAPTCHA_ERROR',
     OAUTH_ERROR = 'OAUTH_ERROR'
 }
+
+export type LinkEventListener = (linkEvent: LinkEvent) => void
+
```
Comment (amytang0) [inline types/Types.d.ts:320]: nit: extra line

**Review by amytang0 — APPROVED** [review]
LGTM

**Comment by amytang0** [comment]
This should wait until 6.0.3 is published ~Nov 3

**Pull request merged**